scio-google-cloud-platform/src/it/scala/com/spotify/scio/bigquery/BigQueryIT.scala

```diff
@@ -93,16 +92,33 @@ class BigQueryIT extends AnyFlatSpec with Matchers {
         "da" -> da.toString
       )
 
-    val inData = Seq(
-      gsod(55.0, 2022, 2, 11),
-      gsod(50.7, 2022, 1, 13),
-      gsod(50.5, 2022, 1, 13),
-      gsod(9999.9, 2022, 1, 1)
+    val suffixData = Map(
+      "21" -> Seq(
+        gsod(54.2, 2021, 6, 30),
+        gsod(53.2, 2021, 6, 22),
+        gsod(53.1, 2021, 7, 1),
+        gsod(9999.9, 2021, 1, 1)
+      ),
+      "20" -> Seq(
+        gsod(54.0, 2020, 6, 8),
+        gsod(52.2, 2020, 7, 31),
+        gsod(52.1, 2020, 7, 30),
+        gsod(9999.9, 2020, 1, 1)
+      ),
+      // not part of the 20s decade. will be ignored
+      "19" -> Seq(
+        gsod(54.5, 2019, 8, 29),
+        gsod(54.4, 2019, 8, 30),
+        gsod(54.4, 2019, 8, 29),
+        gsod(9999.9, 2019, 1, 1)
+      )
     )
 
     val mbq = MockBigQuery()
-    mbq.mockWildcardTable(prefix, suffix).withData(inData)
-    mbq.queryResult(wildcardSqlQuery) should contain only inData.head
+    suffixData.foreach { case (suffix, inData) =>
+      mbq.mockWildcardTable(prefix, suffix).withData(inData)
+    }
+    mbq.queryResult(wildcardSqlQuery) should contain only suffixData("21").head
```

Comment:
I thought BigQuery `BETWEEN` was inclusive, shouldn't the query also contain `suffixData("20")`?

Reply:
the query retrieves the max temperature of the decade (`LIMIT 1`)
The highest one in the test data is in 2021, 1st element

Reply:
Seems like removing `LIMIT 1` would provide a better test?